hypothesis-python/src/hypothesis/strategies/_internal/strategies.py

```diff
@@ -335,7 +335,10 @@ def example(self) -> Ex:
         @given(self)
         @settings(
             database=None,
-            max_examples=100,
+            # generate only a few examples at a time to avoid slow interactivity
+            # for large strategies. The overhead of @given is very small relative
+            # to generation, so a small batch size is fine.
+            max_examples=10,
```

Comment:
unfortunately this is going to skew the distribution due to the small-example cap.  Maybe we should remove the lower bound on that?  But then there's more risk of flaky health checks...

https://github.com/HypothesisWorks/hypothesis/blob/4a796091567d3bbd514a488bcfb36afcaf713774/hypothesis-python/src/hypothesis/internal/conjecture/engine.py#L1005-L1023

Reply:
oof, I forgot about that. Let's avoid changing that cap. 

We could change `example_generating_inner_function` to a raw instantiation of an engine + data and always use an empty prefix, but then we're changing the distribution from "equivalent to what you would have gotten with default `@given`" (including small examples, `generate_mutations_from`, etc) to "pure random". I think that would still be an improvement, but if you disagree then let's just scrap this.

Reply:
though, I have seen several people use `max_examples=10` to cap the runtime of large strategies. I think these users would be surprised to find that they are inadvertently biasing the distribution. That seems like a good argument for removing the cap and seeing how bad the health checks are.

Reply:
I'm inclined to remove the lower bound; health check issues seem less problematic than distorting small-n runs so much.  (Though starting with the minimal example still makes sense imo)